packages/devextreme/js/__internal/ui/splitter/splitter.ts

```diff
@@ -659,14 +668,107 @@ class Splitter extends CollectionWidgetLiveUpdate<Properties> {
       case 'collapsible':
         this._updateResizeHandlesCollapsibleState();
         break;
-      case 'visible':
-        this._invalidate();
+      case 'visible': {
+        super._itemOptionChanged(item, property, value, prevValue);
+
+        this._itemVisibleOptionChanged(item, value as boolean, prevValue as boolean);
         break;
+      }
       default:
         super._itemOptionChanged(item, property, value, prevValue);
     }
   }
 
+  _itemVisibleOptionChanged(item: Item, value: boolean, prevValue: boolean): void {
+    if (Boolean(prevValue ?? true) === Boolean(value ?? true)) {
+      return;
+    }
+
+    const { direction, paneIndex } = this._getDistributionData(item, value);
+
+    this._updateResizeHandles();
+
+    if (paneIndex && paneIndex < 0) { return; }
+
+    this._updateItemsRestrictions();
+
+    const collapsedDelta = this._getCollapseDelta(
+      item,
+      !value,
+      this._panesCacheSizeVisible,
+      direction,
+    );
+
+    this._itemRestrictions.map((pane) => {
+      pane.maxSize = undefined;
+      pane.resizable = undefined;
+
+      return item;
+    });
+
+    this._layout = getNextLayout(
+      this.getLayout(),
+      collapsedDelta,
+      paneIndex,
+      this._itemRestrictions,
+    );
+
+    this._applyStylesFromLayout(this.getLayout());
+    this._updateItemSizes();
+  }
+
+  _getDistributionData(item: Item, value: boolean | undefined): {
+    direction: CollapseExpandDirection;
+    paneIndex: number;
+  } {
+    const itemIndex = this._getIndexByItem(item);
+    const $item = $(this._itemElements()[itemIndex]);
+    const { items = [] } = this.option();
+
+    const isLastVisibleItem = itemIndex >= findLastIndexOfVisibleItem(items);
+    const isLastVisible = this._isLastVisibleItem(itemIndex) || isLastVisibleItem;
+
+    if (value === false) {
+      if (isLastVisible) {
+        return {
+          direction: CollapseExpandDirection.Next,
+          paneIndex: this._getPaneIndexByElement(this._getResizeHandleLeftItem($item)),
+        };
+      }
+
+      return {
+        direction: CollapseExpandDirection.Previous,
+        paneIndex: this._getPaneIndexByElement($item),
+      };
+    } if (isLastVisible) {
```

Comment:
Move to next line, please
It is hard to read 4 returns, perhaps, it would be easier to calculate direction and paneIndex first
Suggestion requires to set default value for `value` to true to work proper with undefined value
```suggestion
    const direction = value === isLastVisible ? CollapseExpandDirection.Previous : CollapseExpandDirection.Next;
    const paneIndex = isLastVisible ? this._getPaneIndexByElement(this._getResizeHandleLeftItem($item)) : this._getPaneIndexByElement($item);
    
    return {
      direction,
      paneIndex,
    };
```

Reply:
I made all conditions separate because it is easier to understand and debug. Both variants are okay for me if the shorter one suits you.

https://github.com/DevExpress/DevExtreme/pull/28876/commits/58a0820fcd17a0422a9d90b12eaee01c5e286602